File: 프로그래머스/위장.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        
        HashMap<String, Integer> map = new HashMap<>();
        
        for(int i = 0 ; i < clothes.length; i++){
            // System.out.println(clothes[i][0] + " " + clothes[i][1]);
            
            if (map.containsKey(clothes[i][1])){
                map.put(clothes[i][1], map.get(clothes[i][1]) + 1);
            }
            else{
                map.put(clothes[i][1], 1);
            }
        }
        System.out.println(map);
        
        int answer = 1;
        
        for(Map.Entry<String, Integer> entry : map.entrySet()){
            answer *= (entry.getValue() + 1);
        }
        
        return answer - 1;
        
    }
}

File: 프로그래머스/위장.py
def solution(clothes):
    
    cloth_dict = dict()
    for cloth, type in clothes:
        if type in cloth_dict:
            cloth_dict[type] += 1
        else:
            cloth_dict[type] = 1
        
    print(cloth_dict)
    
    ans = 1
    for item in cloth_dict:
        ans *= (cloth_dict[item] + 1)
        
    return ans-1